lom/src/pluginmgr/pluginmgr_common/pluginmgr_helper.go

```diff
@@ -833,36 +881,45 @@ func ParseArguments() {
 
 /*
  * Start Plugin Manager - Create Plugin Manager, read each plugin name and its parameters from actions_conf file & Setup each plugin
- */
-func StartPluginManager() error {
+    * Input:
+    *  waittime - 0 for infinite wait,  >0 for wait time in seconds
+    * Output:
+    *  none -
+*/
+func StartPluginManager(waittime time.Duration) error {
     // Create & Start Plugin Manager and do registration with engine
     vclientTx := lomipc.GetClientTx(int(GOLIB_TIMEOUT_DEFAULT))
     vpluginManager := GetPluginManager(vclientTx)
-    lomcommon.LogInfo("Plugin Manager created successfully")
+    if vpluginManager == nil {
+        return LogError("StartPluginManager : Error creating plugin manager")
+    }
+    LogInfo("StartPluginManager : Plugin Manager created successfully")
 
     /* For a particular proc_X, read each plugin name and its parameters from proc_conf.json file &
        Setup each plugin */
     procInfo, err := lomcommon.GetConfigMgr().GetProcsConfig(ProcID)
     if err != nil {
-        return lomcommon.LogError("Error getting proc config for proc %s: %v", ProcID, err)
+        vpluginManager = nil
+        return LogError("StartPluginManager : Error getting proc config for proc %s: %v", ProcID, err)
     }
 
     // TODO: Goutham : Note : Dynamic Plugins is not supported in V1 release. So no code is implemented for dynamic plugins
     for pluginname, plconfig := range procInfo {
-        lomcommon.LogInfo("Initializing plugin %s version %s", pluginname, plconfig.Version)
+        LogInfo("StartPluginManager : Initializing plugin %s version %s", pluginname, plconfig.Version)
         errv := vpluginManager.loadPlugin(pluginname, plconfig.Version)
         if errv != nil {
-            lomcommon.LogError("Error Initializing plugin %s version %s : %v", pluginname, plconfig.Version, errv)
+            LogError("StartPluginManager : Error Initializing plugin %s version %s : %v", pluginname, plconfig.Version, errv)
         } else {
             vpluginManager.pluginMetadata[pluginname].SetPluginStage(plugins_common.PluginStageLoadingSuccess)
-            lomcommon.LogInfo("plugin %s version %s successfully Initialized", pluginname, plconfig.Version)
+            LogInfo("StartPluginManager : plugin %s version %s successfully Initialized", pluginname, plconfig.Version)
         }
     }
 
-    lomcommon.GetGoroutineTracker().Start("StartPluginManager"+lomcommon.GetUUID(),
-        vpluginManager.run())
+    lomcommon.GetGoroutineTracker().Start("StartPluginManager_"+ProcID+lomcommon.GetUUID(),
```

Comment:
We could make this blocking synchronous call.

Reply:
TODO